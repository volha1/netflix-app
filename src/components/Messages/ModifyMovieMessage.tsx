import React, { ReactElement, memo } from 'react';
import CloseBtn from '../CloseBtn/index';
import successIcon from '../../common/assets/svg/success-icon.svg';
import './style.scss';

type ModifyMovieWindowProps = {
  onClose: () => void;
  text: string;
};

const ModifyMovieMessage = ({ onClose, text }: ModifyMovieWindowProps): ReactElement => {
  return (
    <div className="modal-window add-movie">
      <CloseBtn onClose={onClose} />
      <div className="icon-wrapper">
        <img className="menu-icon" src={successIcon} alt="Success" />
      </div>
      <h1 className="title">Congratulations!</h1>
      <p>The movie has been {text} database successfully</p>
    </div>
  );
};

export default memo(ModifyMovieMessage);
