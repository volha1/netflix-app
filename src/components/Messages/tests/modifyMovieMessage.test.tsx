import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import ModifyMovieMessage from '../ModifyMovieMessage';

describe('ModifyMovieMessage test', () => {
  const onClose = jest.fn();

  it('ModifyMovieMessage snapshot', () => {
    const component = renderer.create(<ModifyMovieMessage onClose={onClose} text="" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Right text should be dispayed in message window', () => {
    const text = 'edited in';
    const { getByText } = render(<ModifyMovieMessage onClose={onClose} text={text} />);

    expect(getByText(`The movie has been ${text} database successfully`)).toBeInTheDocument();
  });
});
