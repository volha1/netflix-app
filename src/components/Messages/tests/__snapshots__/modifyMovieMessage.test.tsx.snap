// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModifyMovieMessage test ModifyMovieMessage snapshot 1`] = `
<div
  className="modal-window add-movie"
>
  <div
    className="close-btn"
    data-testid="close-btn"
    onClick={[MockFunction]}
  />
  <div
    className="icon-wrapper"
  >
    <img
      alt="Success"
      className="menu-icon"
    />
  </div>
  <h1
    className="title"
  >
    Congratulations!
  </h1>
  <p>
    The movie has been 
     database successfully
  </p>
</div>
`;
